be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/SchemaVerifier.scala

```diff
@@ -39,14 +39,20 @@ object SchemaVerifier {
 
   private def verifySchema(schema: JsonSchema, jsonString: JsonString): Try[Unit] = {
     // creation of a JsonNode containing the information from the input JSON string
-    val jsonNode: JsonNode = objectMapper.readTree(jsonString)
-
-    // validation of the input, the result is a set of errors (if no errors, the set is empty)
-    // Note: the library is written in Java, thus we convert the Java Set<T> into a Scala Set[T]
-    val errors: Set[ValidationMessage] = schema.validate(jsonNode).asScala.toSet
-    errors match {
-      case _ if errors.isEmpty => Success((): Unit)
-      case _ => Failure(new ProtocolException(errors.mkString("; "))) // concatenate all schema validation errors into one
+    try {
+      //in case of invalid JsonString, we catch the exception thrown by the parser and answer with an error
+      val test = jsonString.parseJson
+
+      val jsonNode: JsonNode = objectMapper.readTree(jsonString)
+      // validation of the input, the result is a set of errors (if no errors, the set is empty)
+      // Note: the library is written in Java, thus we convert the Java Set<T> into a Scala Set[T]
+      val errors: Set[ValidationMessage] = schema.validate(jsonNode).asScala.toSet
+      errors match {
+        case _ if errors.isEmpty => Success((): Unit)
+        case _ => Failure(new ProtocolException(errors.mkString("; "))) // concatenate all schema validation errors into one
+      }
+    } catch {
+      case e: Exception => Failure(new ProtocolException("invalid object detected, JSON object expected"))
```

Comment:
```suggestion
      case e: Exception => Failure(new ProtocolException("Invalid object detected, JSON object expected."))
```

Reply:
(really tiny detail, especially considering that there is no fixed error message grammar throughout the entire code)

Reply:
Understood! Took the comments in consideration